pkg/cache/application.go

```diff
@@ -107,13 +107,13 @@ func NewApplication(appID, queueName, user string, tags map[string]string, sched
 				Src: []string{states.Running},
 				Dst: states.Running},
 			{Name: string(events.ReleaseAppAllocation),
-				Src: []string{states.Failed},
+				Src: []string{states.Failing},
```

Comment:
I think this transition is not correct. Let's say we have an application with 3 allocations. The app will be moved into the Failed state right after removing the first allocation. We want to have the app Failed only when everything is cleared and no allocations and asks are left.